packages/lesswrong/server/migrations/2022-11-08-rehostPostImages.ts

```diff
@@ -0,0 +1,27 @@
+import { registerMigration, forEachDocumentBatchInCollection } from "./migrationUtils";
+import Posts from "../../lib/collections/posts/collection";
+import { convertImagesInPost } from "../scripts/convertImagesToCloudinary";
+
+registerMigration({
+  name: "rehostPostImages",
+  dateWritten: "2022-11-08",
+  idempotent: true,
+  action: async () => {
+    let postCount = 0;
+    let uploadCount = 0;
+    await forEachDocumentBatchInCollection({
+      collection: Posts,
+      batchSize: 100,
+      callback: async (posts) => {
+        const uploadCounts = await Promise.all(
+          // To save cloudinary credits, only convert google docs images since those are the only ones with known problems
+          posts.map(post => convertImagesInPost(post._id, url => url.includes("googleusercontent")))
```

Comment:
I restricted it to only google docs images for old posts because I'm concerned about going over our cloudinary usage limits. This will still be around 3000 images though, and the pricing is mainly by bandwidth so there still a bit of a risk